moz_telemetry/modules/moz_telemetry/normalize.lua

```diff
@@ -192,6 +192,15 @@ function os(name)
     return normalize_os_grammar:match(name)
 end
 
+local normalize_os_version_grammar =
+    l.C(l.R"09"^1 * (l.P"." * l.R"09"^1)^-2)
+
+function os_version(v)
+    if type(v) == "string" then
+        return normalize_os_version_grammar:match(v)
+    else
+        return tostring(v)
```

Comment:
Please convert to a string value before running the 'normalize' grammar to avoid things like `-1` or `nil` leaking into the normalized field.

Something like:
```
vs = tostring(v)
if vs then
  return normalize_os_version_grammar:match(vs)
end
```